docs/vignette_parallel_phylo.md

```diff
@@ -206,43 +227,85 @@ def union_recursion(df, focal="root", merged=None):
+(Since the population in SLiM is ``p1``, the zero-th population will be unused.)
+
 Let's make sure we have the right number of samples in each of the populations
 we specified.
 
 ```{code-cell}
-is_pop = {name: tsu.tables.nodes.population == i+1 for i, name in enumerate(pops)}
-is_present = np.isclose(tsu.tables.nodes.time, 0.0)
-is_sample = np.zeros(tsu.num_nodes)
-is_sample[tsu.samples()] = 1
-present_samples_pop = {pop: np.where(is_pop[pop] + is_sample + is_present == 3)[0] for pop in is_pop}
-for p in is_pop:
-    n_samples = len(present_samples_pop[p]) // 2
-    print(f"Union-ed tree sequence has {n_samples} samples in population {p}, and we specified {df[df.child==p].popsize.item()} individuals in our simulations.")
+for i, name in enumerate(pops):
+    n_samples = len(tsu.samples(i + 1)) // 2
+    print(f"Union-ed tree sequence has {n_samples} samples in population {name},\n"
+          f"\tand we specified {df[df.child==name].popsize.item()} individuals in our simulations.")
 ```
 
-Also, we can look at the trees produced by the simulation, and check for
-disagreements between the species tree and the simulated gene trees (also called ILS).
+Finally, we should recapitate the result,
+in case some trees on the root branch haven't coalesced,
+and write out the result:
+```{code-cell}
+tsu = pyslim.SlimTreeSequence(tsu)
+tsu = tsu.recapitate(recombination_rate=1e-8)
+tsu.dump("final.trees")
+```
 
-For that, I will first simplify the union-ed tree sequence to keep only three
+Now we're done, and can analyse the final tree sequence!
+Just for fun, I'll look at the trees produced by the simulation.
+For instance, we might be curious how often there are
+disagreements between the species tree and the simulated gene trees
+(also called incomplete lineage sorting, or ILS).
+
+To make it possible to look at the trees,
+I will first simplify the union-ed tree sequence to keep only two diploid
 samples per population. Then, I'll recapitate the tree sequence so that we have
 a single root in all the trees.
 
 ```{code-cell}
-subsamples = {p: np.random.choice(s, 3) for p, s in present_samples_pop.items()}
-subs = list(subsamples.values())
-tsus = tsu.simplify(np.concatenate(subs), keep_input_roots=True)
-tsur = pyslim.SlimTreeSequence.load_tables(tsus.tables)
-tsur = tsur.recapitate(recombination_rate=1e-8)
-SVG(tsur.draw_svg("_static/unioned_ts.svg", node_labels={node.id: pops[node.population] for node in tsus.nodes() if not node.time > 0.0}, x_lim=[0,2200]))
+rng = np.random.default_rng(seed=123)
+ind_alive = tsu.individuals_alive_at(0)
+ind_pops = tsu.individual_populations[ind_alive]
+subsample_indivs = [
+    rng.choice(ind_alive[ind_pops == i + 1], 2)
+    for i, _ in enumerate(pops)
+]
+subsample_nodes = [
+    np.concatenate([tsu.individual(i).nodes for i in x])
+    for x in subsample_indivs
+]
+tsus = pyslim.SlimTreeSequence(
+    tsu.simplify(
+        np.concatenate(subsample_nodes),
+        filter_populations=False,
```

Comment:
did this work without the keep_input_rooots? I noticed that we would lose the root nodes without it.

Reply:
I recapitated earlier on instead - I figured that was less confusing.